src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const value = String(item[key]);
+
+    if (!(value in result)) {
+      result[value] = [];
+    }
```

Comment:
You can check like that or use hasOwnProperty
```suggestion
    if (!value[value])) {
      result[value] = [];
    }
```